sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -3674,6 +3674,21 @@ class DataFrameSuite extends QueryTest
         parameters = Map("viewName" -> "AUTHORIZATION"))
     }
   }
+
+  test("SPARK-44846: PushFoldableIntoBranches in complex grouping expressions " +
```

Comment:
Move the test to https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/group-by.sql?